src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenSubmitApplication.java

```diff
@@ -105,5 +107,18 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<C
             .build();
     }
 
+    public void setServiceRequestReferenceForApplicationPayment(CaseData data, long caseId) {
+        final Application application = data.getApplication();
+
```

Comment:
do this creates service request reference always?

Reply:
Yep unless F&P API fails to return one due to an error, in which case an exception will propagate from PaymentService and the user will have to try again later